src/olympia/users/tests/test_models.py

```diff
@@ -795,26 +795,83 @@ def test_blocked_user_login_ip(self):
+        )
+        # Submission remains allowed.
+        assert IPNetworkUserRestriction.allow_submission(request)
+        # Approval is blocked even though it was with a different ip, because
+        # of the user last_login_ip.
+        upload = FileUpload.objects.create(ip_address='192.168.0.2', user=request.user)
+        assert not IPNetworkUserRestriction.allow_auto_approval(upload)
+
+    def test_blocked_approval_while_allowing_submission(self):
+        request = RequestFactory(REMOTE_ADDR='192.168.0.1').get('/')
+        request.user = user_factory(last_login_ip='10.0.0.1')
+        IPNetworkUserRestriction.objects.create(
+            network='192.168.0.0/28', restriction_type=RESTRICTION_TYPES.APPROVAL
+        )
+        # Submission remains allowed.
+        assert IPNetworkUserRestriction.allow_submission(request)
+        # Approval is blocked.
+        upload = FileUpload.objects.create(ip_address='192.168.0.1', user=request.user)
+        assert not IPNetworkUserRestriction.allow_auto_approval(upload)
 
 
 class TestDisposableEmailDomainRestriction(TestCase):
     def test_email_allowed(self):
         DisposableEmailDomainRestriction.objects.create(domain='bar.com')
         request = RequestFactory().get('/')
         request.user = user_factory(email='bar@foo.com')
-        assert DisposableEmailDomainRestriction.allow_request(request)
+        assert DisposableEmailDomainRestriction.allow_submission(request)
 
     def test_email_domain_blocked(self):
         DisposableEmailDomainRestriction.objects.create(domain='bar.com')
         request = RequestFactory().get('/')
         request.user = user_factory(email='foo@bar.com')
-        assert not DisposableEmailDomainRestriction.allow_request(request)
+        assert not DisposableEmailDomainRestriction.allow_submission(request)
 
     def test_user_somehow_not_authenticated(self):
         request = RequestFactory().get('/')
         request.user = AnonymousUser()
-        assert not DisposableEmailDomainRestriction.allow_request(request)
+        assert not DisposableEmailDomainRestriction.allow_submission(request)
+
+    def test_blocked_approval(self):
+        request = RequestFactory().get('/')
+        request.user = user_factory(email='foo@bar.com')
+        DisposableEmailDomainRestriction.objects.create(
+            domain='bar.com', restriction_type=RESTRICTION_TYPES.APPROVAL
+        )
+        # Submission remains allowed.
+        assert DisposableEmailDomainRestriction.allow_submission(request)
+        # Approval is blocked.
+        upload = FileUpload.objects.create(ip_address='192.168.0.1', user=request.user)
+        assert not DisposableEmailDomainRestriction.allow_auto_approval(upload)
+
+    def test_allowed_approval(self):
+        request = RequestFactory().get('/')
+        request.user = user_factory(email='foo@bar.com')
+        DisposableEmailDomainRestriction.objects.create(domain='bar.com')
```

Comment:
Nit: I realize that this shows up in a number of other places, but for some reason it hit me when reading this test: It seems a bit unintuitive that adding a restriction by default creates a submission restriction, and that in order to create an approval restriction one has to specify `restriction_type=RESTRICTION_TYPES.APPROVAL`. I know it's convenient, but it seems like it would be better/safer to always require a `restriction_type` when creating a restriction so it's obvious and explicit what's being done.

Reply:
Yeah, I did that for backwards-compatibility, and was too lazy to go back and change all the existing tests